sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1332,4 +1430,36 @@ public synchronized Throwable fillInStackTrace() {
     }
 
     // endregion
+
+    // TODO: remove when we are confident of RNTBD OOM bug
+    @SuppressWarnings("unchecked")
+    private ScheduledFuture<?> startMonitoring() {
+        return monitoringRntbdChannelPool.scheduleAtFixedRate(() -> {
+            int i = getTaskCount();
+            if (isInterestingEndpoint()) {
+                logger.debug("{} total number of tasks on the executor [{}], connecting [{}], acquiredChannel [{}], availableChannel [{}], pending acquisition [{}]",
+                    this.hashCode(), i, connecting.get(), acquiredChannels.size(), availableChannels.size(), pendingAcquisitions.size());
```

Comment:
In my runs instrumented to include the remoteendpoint and it was super useful too.

Reply:
Added